PP-4617 Build scripts

kbottla opened the pull request:
## WHAT
- Shell scripts to generate static API docs and publish to PaaS 
- Added Jenkins file around `.sh` scripts

kbottla committed:
PP-4617 Build scripts

heathd reviewed (dismissed):
Looks good

(not a blocker) might be worth running shellcheck on it since Russell was trying to do that on all repos

kbottla committed:
PP-4617 Replaced `which` and removed unused variables

Changes as per `shellscript` check suggestions
- Replaced `which` with `command -v'
- Removed unused variables
